### PR TITLE
[0.21] wallet: Avoid requesting fee rates multiple times during coin selection

### PR DESCRIPTION
**Pull request by achow101**
Backport of #21083

### CODE REVIEW

File: src/wallet/test/coinselector_tests.cpp [hunk 1]
```diff
@@ -300,7 +300,7 @@ BOOST_AUTO_TEST_CASE(bnb_search_test)
         CCoinControl coin_control;
```
Comment (MarcoFalke) [inline src/wallet/test/coinselector_tests.cpp:300]: Any reason to not rename in line 269?

**Review by maflcko — COMMENTED** [review]
cherry-pick-only ACK 4b7d19d456a3a1e54bc6f24d4b4e7e8f417896df 🐒

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

cherry-pick-only ACK 4b7d19d456a3a1e54bc6f24d4b4e7e8f417896df 🐒
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUil9wwAvjp9CweUIZcOEI6yjlm/XfbmPfpsNi20feQmD0jg/crBv4zwIRzdAY4L
zuzIhrJ9eG4ylMzQviuAtTS2UBx7BhQCBrMY3ZCNqZSS9m49zT/uhXdlXYXVacIG
prL/i1oL5J77jgxCrt3+zVC6XfO1BHXk8GgA2vWnIpI2MbNC3xS6ulZ2mbTosjsy
QlMv/0+H1qWLmK8Ce6t2S4EszyMHpsiatmIgJsLboIM1NpnwLPoXP7PFbJWW8wEx
eGe7yKd1UldUe6ryu9DOdcWsWNyZzHu9RHVlu5Q9UqUGF0CoXrdLQbC+kX1KcSE6
T0wMNdCN1udoDT3+uUxAtZJsBFbLKqQwhtZBvfHueZrVgAP5fIb9rABMvlBR2jOF
KGw287OdGzh46r/A+GC4sfZRRtol0gR9vr6TlfwU3m+tWO7LfO2YvdjQO6oX5YJS
9m1o4i/f8PEcae1L3SSTiGpZ92mcvNolk3Q+Yr9JfEvaGTmH5m4Rj00zJ7lG4pNG
GpufZO4/
=kp7c
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `03e63e4b18da74fc9ba5be4a98a589d33a69777e74b0004659c5bb473a8cb037  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e89294010803e63e4b18da74fc9ba5be4a98a589d33a69777e74b0004659c5bb473a8cb037f0103e7dc1ab4bba376a8df3b9bb2623d29008fff01002c1a9545d74787fabaafa60184e114e08f1046065b477f00899a45996e24322080083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff0100f4802c3ab29037cf21c7d12904f990f08f1046065b477f00824af23797f2173650083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6dfff01020e4c849a17529540b289983a1e1ea7508f020ab10155ed03f0f804e3d1c84ad067882c20a8fc9c4e346e175914f6e0d95f01608f0206a1fed972c3c283a512531d3b20f72fbd1ee391de560ff4ae0a327ab6f02c05608f1046065a69ef0087a911e8f25ad0dbc0083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6df010ba1e9824210e7f05d81cf437fcff794c08f020fd2650c0dfccddff917ab38c5971d33366bfc0d28c5335959f243a043127ef5108f02045306e754dc2666c1040eb72cddf57af2463c23e2581480f475c42e7d317705f08f1046065b478f0089caf3a2c2fcbda7a0083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267)

</details>

File: src/wallet/test/coinselector_tests.cpp (hunk 1 above)
Comment (achow101) [reply src/wallet/test/coinselector_tests.cpp:300]: Missed it when resolving conflicts. Done.

**Comment by MarcoFalke** [comment]
cherry-pick-only re-ACK d61fb07da7c12e4a1f68cf645f32d563a657a506 🔙

<details><summary>Show signature and timestamp</summary>

Signature:

```
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA512

cherry-pick-only re-ACK d61fb07da7c12e4a1f68cf645f32d563a657a506 🔙
-----BEGIN PGP SIGNATURE-----

iQGzBAEBCgAdFiEE+rVPoUahrI9sLGYTzit1aX5ppUgFAlwqrYAACgkQzit1aX5p
pUgBiwv+PHpjTYKRtUK3v9bYTZ2I0v6hzlydy0CNQAAm1XgJpy0moxairts8I8PL
Gi2dFUw7Utu/dFrucFjnr5SZjK7DyvehGnDSYuLjbNmQqinccmB02ulyrsheuGlT
izIpiku1EjqgBwYpyOg8qq8eAdQ/Ns9CJvnR+LUkXLhkkGr9YE44TcMDFCqJQKEb
y+g6V8J/zTnEnPiUmzeNYMVSTXhfROsblX3jFklb4AyjcvwqJExdq17vGq1aGQ3K
iC+vTmjaXzsliH+Y8x/MtRpf6Z3w8kEw09qGYD9lwSzFD6e0RhrbsEDePdhY/A+b
hv+Xtw25E4SFwpK88AOIhRowOzx6YdV9kO/o5VtaR6a0goqF8r9IF8GefJPzf6Z2
F5xo15FNz08k3sVpzwcaoZY275amIwVi51AoQwBSk4a0WmhQU/gbwFZ0x2puJ+uw
2xUw6VqD3QjGMQA+IefHnfmCULH7iM9nrzMgnxLG9J5Q1aAmHAo0I4DlM8DKolDW
aRM7tnY3
=Wqy6
-----END PGP SIGNATURE-----
```

[Timestamp of file with hash `da605e100922a3feb0ef43fceeae52979a9cfa9e2e57a77c72f8deef904034d7  -`](https://opentimestamps.org/info/?004f70656e54696d657374616d7073000050726f6f6600bf89e2e884e892940108da605e100922a3feb0ef43fceeae52979a9cfa9e2e57a77c72f8deef904034d7f0102be487ea87290da32bc0b5ab3094a09008fff01075631042f8d2bdb88e9003d43eadc1d908f1046065f914f008dc066522bab9956e0083dfe30d2ef90c8e232268747470733a2f2f6274632e63616c656e6461722e636174616c6c6178792e636f6dfff0108b81139aff8557366c3a5b0edb365bed08f1046065f914f008699c6b0b5c9c617b0083dfe30d2ef90c8e2c2b68747470733a2f2f626f622e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267fff010e4201c29ff848c1e6543b0cccbc78adf08f1206fb1b764fa2fb36bdc6fdf04ede30c337512a3c1639e8d268e82f9747359d8f208f1046065f913f0083a30d89c1da934710083dfe30d2ef90c8e2e2d68747470733a2f2f616c6963652e6274632e63616c656e6461722e6f70656e74696d657374616d70732e6f7267f010feedbcf1dcbfc12cd15bce1ecc59217e08f1205375262a7e189cfbd99562c687cc8f7351cc34e66c8fc7dcd2d164bffbad4f9308f1046065eb3bf008f3d1754acc163f590083dfe30d2ef90c8e292868747470733a2f2f66696e6e65792e63616c656e6461722e657465726e69747977616c6c2e636f6d)

</details>

**Comment by instagibbs** [comment]
utACK https://github.com/bitcoin/bitcoin/pull/21520/commits/d61fb07da7c12e4a1f68cf645f32d563a657a506

**Comment by fjahr** [comment]
Code review ACK d61fb07da7c12e4a1f68cf645f32d563a657a506

**Pull request merged**